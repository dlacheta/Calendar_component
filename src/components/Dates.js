import React from "react";
import DateForm from "./DateForm";
import PropTypes from "prop-types";

const Dates = ({
  checkInVisible,
  checkOutVisible,
  toogleCheckIn,
  toogleCheckOut,
  formDate,
  nextMonth,
  previousMonth,
  availableDates,
  pickCheckIn,
  pickCheckOut,
  checkInDate,
  checkOutDate,
  onChangeCheckIn,
  onChangeCheckOut
}) => (
  <div className="dates">
    <span className="small_text">Dates</span>
    <div className="date_inputs">
      <input
        className={checkInVisible ? "input_green" : null}
        placeholder="Check In"
        onClick={toogleCheckIn}
        value={checkInDate ? checkInDate : ""}
        onChange={onChangeCheckIn}
      />
      <i className="material-icons">arrow_forward</i>
      <input
        className={checkOutVisible ? "input_green" : null}
        placeholder="Check Out"
        onClick={toogleCheckOut}
        value={checkOutDate ? checkOutDate : ""}
        onChange={onChangeCheckOut}
      />
    </div>
    {checkInVisible ? (
      <div className="check_in_form">
        <DateForm
          formDate={formDate}
          nextMonth={nextMonth}
          previousMonth={previousMonth}
          availableDates={availableDates}
          pickCheckIn={pickCheckIn}
          pickCheckOut={pickCheckOut}
          checkInVisible={checkInVisible}
        />
      </div>
    ) : null}
    {checkOutVisible ? (
      <div className="check_out_form">
        <DateForm
          formDate={formDate}
          nextMonth={nextMonth}
          previousMonth={previousMonth}
          availableDates={availableDates}
          pickCheckIn={pickCheckIn}
          pickCheckOut={pickCheckOut}
          checkInVisible={checkInVisible}
        />
      </div>
    ) : null}
  </div>
);

Dates.propTypes = {
  checkInVisible: PropTypes.bool.isRequired,
  checkOutVisible: PropTypes.bool.isRequired,
  toogleCheckIn: PropTypes.func.isRequired,
  toogleCheckOut: PropTypes.func.isRequired,
  formDate: PropTypes.object.isRequired,
  nextMonth: PropTypes.func.isRequired,
  previousMonth: PropTypes.func.isRequired,
  availableDates: PropTypes.arrayOf(PropTypes.string),
  pickCheckIn: PropTypes.func.isRequired,
  pickCheckOut: PropTypes.func.isRequired,
  checkInDate: PropTypes.string,
  checkOutDate: PropTypes.string,
  onChangeCheckIn: PropTypes.func.isRequired,
  onChangeCheckOut: PropTypes.func.isRequired
};

Dates.defaultProps = {
  availableDates: [],
  checkInDate: "",
  checkOutDate: ""
};

export default Dates;
