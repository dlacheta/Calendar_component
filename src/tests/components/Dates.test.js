import React from "react";
import { shallow } from "enzyme";
import Dates from "../../components/Dates";

it("Dates component renders correctly", () => {
  const toogleCheckIn = jest.fn();
  const toogleCheckOut = jest.fn();
  const nextMonth = jest.fn();
  const previousMonth = jest.fn();
  const pickCheckIn = jest.fn();
  const pickCheckOut = jest.fn();
  const onChangeCheckIn = jest.fn();
  const onChangeCheckOut = jest.fn();
  const checkInVisible = false;
  const checkOutVisible = false;
  const checkInDate = "";
  const checkOutDate = "";
  const formDate = new Date();
  const availableDates = [
    "4/12/2018",
    "5/12/2018",
    "6/12/2018",
    "7/12/2018",
    "8/12/2018",
    "9/12/2018",
    "10/12/2018",
    "11/12/2018",
    "12/12/2018",
    "13/12/2018",
    "14/12/2018",
    "15/12/2018",
    "16/12/2018",
    "17/12/2018",
    "18/12/2018",
    "19/12/2018"
  ];
  const wrapper = shallow(
    <Dates
      toogleCheckIn={toogleCheckIn}
      toogleCheckOut={toogleCheckOut}
      nextMonth={nextMonth}
      previousMonth={previousMonth}
      pickCheckIn={pickCheckIn}
      pickCheckOut={pickCheckOut}
      onChangeCheckIn={onChangeCheckIn}
      onChangeCheckOut={onChangeCheckOut}
      checkInVisible={checkInVisible}
      checkOutVisible={checkOutVisible}
      formDate={formDate}
      availableDates={availableDates}
      checkInDate={checkInDate}
      checkOutDate={checkOutDate}
    />
  );

  expect(wrapper).toMatchSnapshot();
});
