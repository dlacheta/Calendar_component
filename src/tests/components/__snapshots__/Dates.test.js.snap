// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dates component renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Dates
    availableDates={
      Array [
        "4/12/2018",
        "5/12/2018",
        "6/12/2018",
        "7/12/2018",
        "8/12/2018",
        "9/12/2018",
        "10/12/2018",
        "11/12/2018",
        "12/12/2018",
        "13/12/2018",
        "14/12/2018",
        "15/12/2018",
        "16/12/2018",
        "17/12/2018",
        "18/12/2018",
        "19/12/2018",
      ]
    }
    checkInDate=""
    checkInVisible={false}
    checkOutDate=""
    checkOutVisible={false}
    formDate={2018-12-24T12:33:08.013Z}
    nextMonth={[MockFunction]}
    onChangeCheckIn={[MockFunction]}
    onChangeCheckOut={[MockFunction]}
    pickCheckIn={[MockFunction]}
    pickCheckOut={[MockFunction]}
    previousMonth={[MockFunction]}
    toogleCheckIn={[MockFunction]}
    toogleCheckOut={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <span
          className="small_text"
        >
          Dates
        </span>,
        <div
          className="date_inputs"
        >
          <input
            className={null}
            onChange={[MockFunction]}
            onClick={[MockFunction]}
            placeholder="Check In"
            value=""
          />
          <i
            className="material-icons"
          >
            arrow_forward
          </i>
          <input
            className={null}
            onChange={[MockFunction]}
            onClick={[MockFunction]}
            placeholder="Check Out"
            value=""
          />
        </div>,
        null,
        null,
      ],
      "className": "dates",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Dates",
          "className": "small_text",
        },
        "ref": null,
        "rendered": "Dates",
        "type": "span",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <input
              className={null}
              onChange={[MockFunction]}
              onClick={[MockFunction]}
              placeholder="Check In"
              value=""
            />,
            <i
              className="material-icons"
            >
              arrow_forward
            </i>,
            <input
              className={null}
              onChange={[MockFunction]}
              onClick={[MockFunction]}
              placeholder="Check Out"
              value=""
            />,
          ],
          "className": "date_inputs",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": null,
              "onChange": [MockFunction],
              "onClick": [MockFunction],
              "placeholder": "Check In",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "arrow_forward",
              "className": "material-icons",
            },
            "ref": null,
            "rendered": "arrow_forward",
            "type": "i",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": null,
              "onChange": [MockFunction],
              "onClick": [MockFunction],
              "placeholder": "Check Out",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
        ],
        "type": "div",
      },
      null,
      null,
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <span
            className="small_text"
          >
            Dates
          </span>,
          <div
            className="date_inputs"
          >
            <input
              className={null}
              onChange={[MockFunction]}
              onClick={[MockFunction]}
              placeholder="Check In"
              value=""
            />
            <i
              className="material-icons"
            >
              arrow_forward
            </i>
            <input
              className={null}
              onChange={[MockFunction]}
              onClick={[MockFunction]}
              placeholder="Check Out"
              value=""
            />
          </div>,
          null,
          null,
        ],
        "className": "dates",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Dates",
            "className": "small_text",
          },
          "ref": null,
          "rendered": "Dates",
          "type": "span",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <input
                className={null}
                onChange={[MockFunction]}
                onClick={[MockFunction]}
                placeholder="Check In"
                value=""
              />,
              <i
                className="material-icons"
              >
                arrow_forward
              </i>,
              <input
                className={null}
                onChange={[MockFunction]}
                onClick={[MockFunction]}
                placeholder="Check Out"
                value=""
              />,
            ],
            "className": "date_inputs",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": null,
                "onChange": [MockFunction],
                "onClick": [MockFunction],
                "placeholder": "Check In",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "arrow_forward",
                "className": "material-icons",
              },
              "ref": null,
              "rendered": "arrow_forward",
              "type": "i",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": null,
                "onChange": [MockFunction],
                "onClick": [MockFunction],
                "placeholder": "Check Out",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
          ],
          "type": "div",
        },
        null,
        null,
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
